homeassistant/components/withings/common.py

```diff
@@ -0,0 +1,184 @@
+        self._slug = slugify(profile)
+
+        self._measures = None
+        self._sleep = None
+        self._sleep_summary = None
+
+        self.sleep_summary_last_update_parameter = None
+
+    @property
+    def profile(self) -> str:
+        """Get the profile."""
+        return self._profile
+
+    @property
+    def slug(self) -> str:
+        """Get the slugified profile the data is for."""
+        return self._slug
+
+    @property
+    def api(self):
+        """Get the api object."""
+        return self._api
+
+    @property
+    def measures(self):
+        """Get the current measures data."""
+        return self._measures
+
+    @property
+    def sleep(self):
+        """Get the current sleep data."""
+        return self._sleep
+
+    @property
+    def sleep_summary(self):
+        """Get the current sleep summary data."""
+        return self._sleep_summary
+
+    @staticmethod
+    def print_service_unavailable():
+        """Print the service is unavailable (once) to the log."""
+        if WithingsDataManager.service_available is not False:
+            _LOGGER.error(
+                "Looks like the service is not available at the moment"
+            )
+            WithingsDataManager.service_available = False
+            return True
+
+    @staticmethod
+    def print_service_available():
+        """Print the service is available (once) to to the log."""
+        if WithingsDataManager.service_available is not True:
+            _LOGGER.info("Looks like the service is available again")
+            WithingsDataManager.service_available = True
+            return True
+
+    @staticmethod
+    async def async_call(function, is_first_call=True):
+        """Call an api method and handle the result."""
+        try:
+            _LOGGER.debug("Running call.")
+            result = await function()
+            WithingsDataManager.print_service_available()
+            return result
+
+        except TokenUpdated:
+            WithingsDataManager.print_service_available()
+            if not is_first_call:
+                raise ServiceError(
+                    "Stuck in a token update loop. This should never happen"
+                )
+
+            _LOGGER.info("Token updated, re-running call.")
+            return await WithingsDataManager.async_call(function, False)
+
+        except MissingTokenError as ex:
+            raise NotAuthenticatedError(ex)
+
+        except Exception as ex:  # pylint: disable=broad-except
+            # Service error, probably not authenticated.
+            if NOT_AUTHENTICATED_ERROR.match(str(ex)):
+                raise NotAuthenticatedError(ex)
+
+            # Probably a network error.
+            WithingsDataManager.print_service_unavailable()
+            raise PlatformNotReady(ex)
+
+    @Throttle(const.SCAN_INTERVAL)
+    async def async_check_authenticated(self):
+        """Check if the user is authenticated."""
+        async def function():
```

Comment:
Why are these functions coroutines? They don't seem to await anything. If we don't need them to be coroutines, it's better to let them be regular callbacks.